src/acl/Ip.cc

```diff
@@ -471,13 +374,64 @@ acl_ip_data::FactoryParse(const char *t)
     return q;
 }
 
+/// handles special ACL data parameters that apply to the whole ACLIP object
+/// \returns true if input token is such a special parameter
+bool
+ACLIP::parseGlobal(const char * const token)
+{
+    // "all" matches entire Internet
+    if (strcmp(token, "all") == 0) {
+        debugs(28, 8, "found " << token);
+        matchAnyIpv4 = true;
+        matchAnyIpv6 = true;
+        // TODO: Ignore all other ACL data parameters, with a once/ACL warning.
+        return true;
+    }
+
+    // "ipv4" matches IPv4 Internet
+    if (strcmp(token, "ipv4") == 0) {
+        debugs(28, 8, "found " << token);
+        matchAnyIpv4 = true;
+        // TODO: Ignore all IPv4 data parameters, with a once/ACL warning.
+        return true;
+    }
+
+    // "ipv4" matches IPv6 Internet
+    if (strcmp(token, "ipv6") == 0) {
+        debugs(28, 8, "found " << token);
+        matchAnyIpv6 = true;
+        // TODO: Ignore all IPv6 data parameters, with a once/ACL warning.
+        return true;
+    }
+
+    /* Detect some old broken strings equivalent to 'all'.
+     * treat them nicely. But be loud until its fixed.  */
+    if (strcmp(token, "0/0") == 0 ||
+            strcmp(token, "0.0.0.0/0") == 0 ||
+            strcmp(token, "0.0.0.0/0.0.0.0") == 0 ||
+            strcmp(token, "0.0.0.0-255.255.255.255") == 0 ||
+            strcmp(token, "0.0.0.0-0.0.0.0/0") == 0) {
+
+        debugs(28,DBG_CRITICAL, "ERROR: '" << token << "' needs to be replaced by the term 'all'.");
+        debugs(28,DBG_CRITICAL, "SECURITY NOTICE: Overriding config setting. Using 'all' instead.");
```

Comment:
I left this [2009](https://github.com/squid-cache/squid/commit/7764e92dbe388e6152b559910d4769d8c737b19f) override unchanged. It should probably be removed (with /0 addresses converted into a fatal configuration error?), but that felt like an out of scope change.

Reply:
Are we sure it's a good idea to replace the config instead of aborting and letting the admin update their settings?
The replaced config also matches all ipv6, which may or may not be what the admin intends

Reply:
> Are we sure it's a good idea to replace the config instead of aborting and letting the admin update their settings?

FWIW, IMO, replacing the config in this particular case (including the fact that this is master/v7 code) is bad. That is why I said that this official 2009 workaround code should be removed or replaced (in a dedicated PR).

> The replaced config also matches all ipv6, which may or may not be what the admin intends

Yes, that is one of this official code problems.

Just to avoid misunderstanding: You are asking a question about official code that this PR does not change in any material way (only moves and synchronizes it with the `all` mechanism changes done in this PR). As I said, I believe this code should be removed or replaced with a different (and probably fatal) error, but all of those changes are outside _this_ PR scope.

If you prefer, I can return this code to its original place and do not update it to use the PR-adjusted `all` mechanism, but that feels wrong to me because this old code explicitly says `using 'all'` _and_ has the effect of using `all`.

Reply:
If the behaviour is what is currently shipped, then there is no reason to change here.

I would support a different PR changing the behaviour from warn-and-change to warn-and-exit